app/models/job.rb

```diff
@@ -38,20 +38,30 @@ class << self
   STATES.each do |s|
     define_method("#{s.downcase}?") { self.state == s }
   end
+
+  # TODO: Check if the complexity in the complete? method is required
   def completed?
-    job_type == 'ARRAY_JOB' ?
-      array_tasks.all?(&:completed?) :
+    if job_type == 'ARRAY_JOB'
+      return false unless task_registry.finished?
+      task_registry.past_tasks.all?(&:complete?)
+    else
       self.state == 'COMPLETED'
+    end
   end
 
-  JOB_TYPES = %w( JOB ARRAY_JOB ARRAY_TASK ).freeze
+  JOB_TYPES = %w( JOB ARRAY_JOB ).freeze
 
   # The index of the task inside the array job.  Only present for ARRAY_TASKS.
   attr_accessor :array_index
 
   # A reference to the ARRAY_JOB.  Only present for ARRAY_TASKS.
   attr_accessor :array_job
 
+  # The node an individual task is ran on. This is not used by the "job" variety of Job
+  # NOTE: Do not expose this publicly! It is solely used by the scheduler to track
+  #       which task is running where
+  attr_accessor :task_node
```

Comment:
I don't think that this is used anywhere.  Perhaps its left over from a previous iteration.  If it is still used, but only by `Task`s, it would be better to define it on `Task`.

Reply:
Good spot, removed: https://github.com/openflighthpc/flight-scheduler-controller/pull/12/commits/fd44c472fb81b692adc82cc796ab05ce06682bfd